src/toil/jobStores/awsJobStore.py

```diff
@@ -234,117 +249,125 @@ def delete(self, jobStoreID):
-        version = self._getFileVersion(jobStoreFileID)
-        if version is None: raise NoSuchFileException(jobStoreFileID)
-        log.debug("Reading version %s of file %s", version, jobStoreFileID)
-        with self._downloadStream(jobStoreFileID, version, self.files) as readable:
+        fileInfo = FileInfo.load(self,jobStoreFileID)
+        if fileInfo is None: raise NoSuchFileException(jobStoreFileID)
+        version = fileInfo.version if fileInfo.version else "inlined"
+        log.debug("Reading version %s of fileInfo %s", version, jobStoreFileID)
+        with self._downloadStream(fileInfo) as readable:
             yield readable
 
     @contextmanager
     def readSharedFileStream(self, sharedFileName, isProtected=True):
         assert self._validateSharedFileName(sharedFileName)
         jobStoreFileID = self._newFileID(sharedFileName)
-        version = self._getFileVersion(jobStoreFileID)
-        if version is None: raise NoSuchFileException(jobStoreFileID)
-        log.debug("Read version %s from shared file %s (%s)",
-                  version, sharedFileName, jobStoreFileID)
-        with self._downloadStream(jobStoreFileID, version, self.files,
-                                  encrypted=isProtected) as readable:
+        fileInfo = FileInfo.load(self, jobStoreFileID, isProtected)
+        if fileInfo is None: raise NoSuchFileException(jobStoreFileID)
+        log.debug("Read version %s from shared fileInfo %s (%s)",
+                  fileInfo, sharedFileName, jobStoreFileID)
+        with self._downloadStream(fileInfo, encrypted=isProtected) as readable:
             yield readable
 
     def deleteFile(self, jobStoreFileID):
-        version, bucket = self._getFileVersionAndBucket(jobStoreFileID)
-        if bucket:
+        fileInfo = FileInfo.load(self,jobStoreFileID)
+        if fileInfo is not None:
             for attempt in retry_sdb():
                 with attempt:
-                    if version:
-                        self.versions.delete_attributes(jobStoreFileID,
-                                                        expected_values=['version', version])
+                    if fileInfo.version:
+                            self.versions.delete_attributes(jobStoreFileID,
+                                                        expected_values=['version', fileInfo.version])
+                            fileInfo.bucket.delete_key(key_name=jobStoreFileID, version_id=fileInfo.version)
                     else:
                         self.versions.delete_attributes(jobStoreFileID)
 
-            bucket.delete_key(key_name=jobStoreFileID, version_id=version)
-            if version:
-                log.debug("Deleted version %s of file %s", version, jobStoreFileID)
-            else:
-                log.debug("Deleted unversioned file %s", jobStoreFileID)
+
+                if fileInfo.version:
+                    log.debug("Deleted version %s of fileInfo %s", fileInfo.version, jobStoreFileID)
```

Comment:
Please be more thorough with search and replace. Especially with common strings like "file". How many times have I pointed this out now?